lib/meadow_web/controllers/auth_controller.ex

```diff
@@ -1,19 +1,48 @@
 defmodule MeadowWeb.AuthController do
   use MeadowWeb, :controller
-  alias Ueberauth.Strategy.OpenAM
 
-  def login(conn, _params) do
+  plug Ueberauth
+  alias(Meadow.Accounts)
+
+  def callback(%{assigns: %{ueberauth_failure: _fails}} = conn, _params) do
+    referer =
+      conn
+      |> extract_referer()
+
     conn
-    |> OpenAM.handle_request!()
+    |> redirect(external: referer)
   end
 
-  def callback(conn, _params) do
-    conn = OpenAM.handle_callback!(conn)
+  def callback(%{assigns: %{ueberauth_auth: auth}} = conn, _params) do
+    referer =
+      conn
+      |> extract_referer()
+
+    case Accounts.user_from_auth(auth) do
+      {:ok, user} ->
+        conn
+        |> put_session("current_user", user)
+        |> configure_session(renew: true)
+        |> redirect(external: referer)
+
+      {:error, _reason} ->
+        conn
+        |> redirect(external: referer)
+    end
+  end
 
-    body = OpenAM.uid(conn)
+  def logout(conn, _params) do
+    conn
+    |> delete_session(:current_user)
```

Comment:
For consistency's sake, can you use either `"current_user"` (as on line 24) or `:current_user` in both places? I prefer atoms wherever possible, but whichever.

Would atoms work for header names (`:content_type`, `:referer`) also?

Reply:
the answer seems to be no on referer but I made the other change.